### ISSUE TITLE
File tracking failure occurs when a different file with the same name directly replace the original one.

### ISSUE THREAD

**Issue by ZYMelaii** [sumatrapdfreader/sumatrapdf#2867]
I noticed that when a file with same name was moved to cover the on-tracked file, SumatraPDF did not regard it as an update. However, under any other cases, the tracking procedure works fine. It really fuzied me as my xelatex driver always generates new PDF via a move-and-cover method and then, the viewer won't be refreshed.

**Comment by ZYMelaii** [comment]
I consider it as an syscall limitation. But if it is possible to fix the bug internally, it can be a great deal!

**Comment by GitHubRulesOK** [comment]
Depends how a file is rebuilt, traditionally LaTeX.pdf under 10 MB were auto refreshed on re compile, the limit should now be 32MB there are system factors that can delay the signal to refresh such as other file interference by AntiVirus or File Syncronisation outside SumatraPDF control.

Silly example 
externally file was changed (same filename) by replace with icon in fresh position and SumatraPDF auto refreshed simply on that external reposition.


![image](https://user-images.githubusercontent.com/12754913/184669375-3e010fbc-9d14-4c98-b35a-f465bda4405c.png)
![image](https://user-images.githubusercontent.com/12754913/184670212-5c026f6b-50ac-4c8f-b4d6-53abc030ae99.png)

**Comment by kjk** [comment]
Can you provide a reproduction steps, preferably one that doesn't use xelatex but some simple way to reproduce the problem?

**Comment by zymelaii** [comment]
> Can you provide a reproduction steps, preferably one that doesn't use xelatex but some simple way to reproduce the problem?

Simply generate pdf A under folder T, and PDF B under folder G via any methods (note that B differs from A). Open A in the SumatraPDF then execute shell e.g. `mv -f G/B.pdf T/A.pdf' - no changes in the SumatraPDF after that.

I found only latexmk with xelatex driver sometimes performs like this, while others work well.

**Comment by fuhrmanator** [comment]
Typing "R" (refresh) is a workaround, but I confirm this problem happens in my Quarto builds (latex is used under the hood, but the quarto build process deletes the PDF at the start, and so when it's rebuilt it is technically a new file). 

Note that Quarto has its own PDF previewer via a web interface (it starts up a web server running on localhost and you open the PDF in a browser). So updating works, more or less, but I find it clunky especially when the localhost port changes. For these reasons, I prefer SumatraPDF.

**Comment by mirh** [comment]
This sounds awfully similar to the problem I documented in https://github.com/SublimeText/LaTeXTools/issues/1373. 
It used to work before W10 1709. 

> It really fuzied me as my xelatex driver always generates new PDF via a move-and-cover method and then

That could explain why latex tools are so special and nobody else in the hell seems to notice. 
I wonder if some [slightly](https://github.com/jimbeveridge/readdirectorychanges) [different](https://github.com/tresorit/rdcfswatcherexample) implementations of the same we use here are also affected? 


p.s. FWIW my testcase is running through [tinytex](https://yihui.org/tinytex/#for-r-users) a [thesisdown](https://github.com/ismayc/thesisdown) document in [RStudio](https://posit.co/download/rstudio-desktop/) (which comes with sumatra built-in btw)

**Comment by GitHubRulesOK** [comment]
@kjk 
again I am reminded this issue has tones of the refresh problems encountered by some LaTeX users in recent https://github.com/sumatrapdfreader/sumatrapdf/issues/3774
so is it a common file replacement mis detection ?

**Comment by mirh** [comment]
Isn't #3774 about something bad/undesired happening **after** refresh happens, as opposed to refresh *not happening at all*?

**Comment by changlichun** [comment]
@kjk I notice that moving file (or `mv` command) only causes a `FILE_ACTION_ADDED`  notion. But sumatrapdf do nothing with it. May be let sumatrapdf treat `FILE_ACTION_ADDED` just same as `FILE_ACTION_MODIFIED` in [line](https://github.com/sumatrapdfreader/sumatrapdf/blob/26606ddf47ba8400871b1e09f8131a2959b23a02/src/utils/FileWatcher.cpp#L244)
```
if (notify->Action == FILE_ACTION_ADDED || notify->Action == FILE_ACTION_MODIFIED ||
            notify->Action == FILE_ACTION_RENAMED_NEW_NAME)
```
It seems worked, but I am not sure will this cause any other iusse else.

**Comment by kjk** [comment]
@changlichun makes sense. probably won't break anything. Would you like to make a PR?

**Comment by kjk** [comment]
Re-tested against current master and I can't reproduce this any more, so I'm closing it.

I automated all four ways the file can be replaced under an open document (one-page PDF open in SumatraPDF, replaced by a different one, checking the rendered page after each):

- overwrite the file in place — refreshed
- `rename` another file over it (what latexmk / `mv -f` does) — refreshed
- delete, pause, then create a new file with the same name (what quarto does) — refreshed
- delete, pause, then rename another file in — refreshed

If this still happens for you on the current [pre-release](https://www.sumatrapdfreader.org/prerelease), please reopen with the exact tool chain that writes the file — most likely the remaining cases involve something else holding the file (antivirus, sync client) or a build tool that writes to a different directory than the one being watched.
